### PR TITLE
Implemented socks proxy support for storm/axon HTTP API

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/stormhttp.py [hunk 1]
```diff
@@ -99,15 +102,24 @@ async def _httpRequest(self, meth, url, headers=None, json=None, body=None, ssl_
...
+        async with aiohttp.ClientSession(connector=connector) as sess:
             try:
                 async with sess.request(meth, url, headers=headers, json=json, data=body, **kwargs) as resp:
                     info = {
                         'code': resp.status,
                         'body': await resp.content.read()
                     }
                     return HttpResp(info)
-            except (TypeError, ValueError) as e:
+            except asyncio.CancelledError: # pragma: no cover
```
Comment (jnwatson) [inline synapse/lib/stormhttp.py:120]: PEP 8:  two spaces between source code and end of comment.

**Review by jnwatson — CHANGES_REQUESTED** [review]
left one very minor comment.

**Review by jnwatson — DISMISSED** [review]

File: synapse/cortex.py [hunk 2]
```diff
@@ -834,6 +834,11 @@ class Cortex(s_cell.Cell):  # type: ignore
             'description': 'Logging log level to emit storm logs at.',
             'type': 'integer'
         },
+        'http:proxy': {
+            'description': 'An aiohttp-socks compatible proxy URL to use storm HTTP API.',
+            'type': 'string',
+            'hidecmdl': True,
```
Comment (vEpiphyte) [inline synapse/cortex.py:840]: Why is this set to hidecmdl ?

File: synapse/axon.py [hunk 3]
```diff
@@ -236,7 +238,12 @@ class Axon(s_cell.Cell):
             'type': 'integer',
             'minimum': 1,
             'hidecmdl': True,
-        }
+        },
+        'http:proxy': {
+            'description': 'An aiohttp-socks compatible proxy URL to use in the wget API.',
+            'type': 'string',
+            'hidecmdl': True,
```
Comment (vEpiphyte) [inline synapse/axon.py:245]: Why is this set to hidecmdl ?

File: synapse/cortex.py (hunk 2 above)
Comment (invisig0th) [reply synapse/cortex.py:840]: To avoid config option CLI param clutter, but i could def remove.

**Review by jnwatson — APPROVED** [review]

**Pull request merged**